nixos/modules/services/x11/window-managers/qtile.nix

```
@@ -1,23 +1,63 @@
-{ config, lib, pkgs, ... }:
+{ config, pkgs, lib, ... }:
 
 with lib;
 
 let
   cfg = config.services.xserver.windowManager.qtile;
+  pyEnv = pkgs.python3.withPackages (p: [ cfg.package ] ++ (cfg.extraPackages p));
 in
 
 {
   options.services.xserver.windowManager.qtile = {
     enable = mkEnableOption (lib.mdDoc "qtile");
 
     package = mkPackageOptionMD pkgs "qtile" { };
+
+    configFile = mkOption {
+      type = with types; nullOr path;
+      default = null;
+      example = literalExpression "./your_config.py";
+      description = lib.mdDoc ''
+          Path to the qtile configuration file.
+          If null, $HOME/.config/qtile/config.py will be used.
```

```suggestion
          If null, $XDG_CONFIG_HOME/qtile/config.py will be used.
```

This is how it's stated in qtile's manual